### PR TITLE
Add team/permission checks for harvesting and curation

### PR DESCRIPTION
**Pull request by jpeddicord**
Note the two team names in config. These can be changed; they just have to match teams in the clearlydefined org.

### CODE REVIEW

File: middleware/github.js [hunk 1]
```diff
@@ -54,12 +54,16 @@ module.exports = asyncMiddleware(async (req, res, next) => {
...
 
-    // TEMPORARY: ideally we'd not use this, but PATs as-configured
-    // don't get access to teams. someday switch everyone to personal
-    // oauth apps?
     try {
       teams = await getTeams(client, config.auth.github.org);
     } catch (err) {
+      if (err.code === 404) {
+        console.error('GitHub returned a 404 when trying to read team data. ' +
+          'You probably need to re-configure your CURATION_GITHUB_TOKEN token with the `read:org` scope. (This only affects local development.)');
```
Comment (jeffmcaffer) [inline middleware/github.js]: nice!

File: routes/auth.js [hunk 2]
```diff
@@ -60,7 +60,7 @@ setup.getStrategy = () => {
     clientSecret: config.auth.github.clientSecret,
     // this needs to match the callback url on the oauth app on github
     callbackURL: `${config.endpoints.service}/auth/github/finalize`,
-    scope: ['repo', 'user']
+    scope: ['public_repo', 'read:user', 'read:org']
```
Comment (jeffmcaffer) [inline routes/auth.js]: Nice to scope those down. did you confirm that public_repo gives the user enough perms to open a pull request?

Comment (jpeddicord) [reply routes/auth.js]: Yep, it functioned fine when I last tested it. Latest pull of web UI isn't quite functioning for me at the moment (missing navbar & curation screen non-functioning) but it looks like you were in the middle of something there.

**Comment by jpeddicord** [comment]
@jeffmcaffer I'll let you do the merge since you need to create teams on the clearlydefined org in order for it to work. `harvesters` and `curators` are the team names; I'd recommend creating those and adding current members to them for now.

**Pull request merged**